packages/slidev/node/export.ts

```diff
@@ -89,39 +92,55 @@ export async function exportSlides({
   const page = await context.newPage()
   const progress = createSlidevProgress()
 
-  async function go(no: number, clicks?: string) {
-    progress.update(no)
-
+  async function go(no: number | string, clicks?: string) {
     const path = `${no}?print${withClicks ? '=clicks' : ''}${clicks ? `&clicks=${clicks}` : ''}`
     const url = routerMode === 'hash'
       ? `http://localhost:${port}${base}#${path}`
       : `http://localhost:${port}${base}${path}`
-
     await page.goto(url, {
       waitUntil: 'networkidle',
     })
-    await page.waitForTimeout(timeout)
     await page.waitForLoadState('networkidle')
     await page.emulateMedia({ colorScheme: dark ? 'dark' : 'light', media: 'screen' })
+    // Wait for twitter iframe if we have some embedded tweets
+    const count = await page.locator('.tweet').count()
+    if (count > 0)
+      await page.locator('.tweet iframe').waitFor()
```

Comment:
This is a bit too specific, possible to have a more general solution

Reply:
I would like but that's a fix that is specific for Twitter because of the use of the Twitter API.
The `<Tweet>` component does not render an iframe, it renders a `<div>` with an icon inside.
And without this, Playwright does not wait for the Tweeter component to completely render. (I see the carbon icon in the export)
So in fact I tell Playwright to wait for the twitter <iframe>` element to be present in the DOM (only if I use the `<Tweet>` component).
This does not happen with the `<Youtube>` since it already dislpays the `<iframe>` element.

The only way I think to make this more general it to use some data in the componen.
For example in the `<Tweet>` you can have something like:
```
<template>
  <Transform :scale="scale || 1">
    <div ref="tweet" class="tweet" data-waitfor="iframe">
      <div v-if="!loaded" class="w-30 h-30 my-10px bg-gray-400 bg-opacity-10 rounded-lg flex opacity-50">
        <div class="m-auto animate-pulse text-4xl">
          <carbon:logo-twitter />
        </div>
      </div>
    </div>
  </Transform>
</template>
```
So if we need to wait for something else we can reuse the `data-waitfor` attribute inside another component.
What do you think?

Reply:
Could we generally wait for all iframes?

Reply:
Waiting for all iframes is done on the line after:
```js
const frames = await page.frames()
```
Here we wait for the iframe to be present in the document, before waiting for the iframe to be loaded.
But anyway I will push a reusable solution, in case there is an other component that need to wait for specific DOM elements before printing the page.